File: 15-18/IPC/IPCGuest/app/src/main/java/com/wx/ipcguest/MainActivity.java
package com.wx.ipcguest;

import android.databinding.ViewDataBinding;
import android.view.View;

import com.wx.ipcguest.activity.AidlActivity;
import com.wx.ipcguest.activity.MessengerActivity;
import com.wx.ipcguest.activity.base.BaseActivity;
import com.wx.ipcguest.databinding.ActivityMainBinding;

public class MainActivity extends BaseActivity {

    private ActivityMainBinding binding;

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void setDataBinding(ViewDataBinding binding) {
        this.binding = (ActivityMainBinding) binding;
    }

    @Override
    public void initData() {
        binding.btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openActivity(MessengerActivity.class);
            }
        });
        binding.btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openActivity(AidlActivity.class);
            }
        });
    }

}

File: 15-18/UiMaster/app/src/main/java/com/wx/uimaster/MainActivity.kt
package com.wx.uimaster

import android.databinding.ViewDataBinding
import android.view.View
import com.wx.uimaster.activity.ComplexRecyclerViewActivity
import com.wx.uimaster.activity.VCustomActivity
import com.wx.uimaster.activity.VStyleActivity

import com.wx.uimaster.base.BaseActivity
import com.wx.uimaster.databinding.ActivityMainBinding

class MainActivity : BaseActivity(), View.OnClickListener {

    private lateinit var binding: ActivityMainBinding

    override fun layoutId(): Int {
        return R.layout.activity_main
    }

    override fun setDataBinding(binding: ViewDataBinding) {
        this.binding = binding as ActivityMainBinding
    }

    override fun initData() {
        binding.complexRecyclerview.setOnClickListener(this)
        binding.vLayoutCustom.setOnClickListener(this)
        binding.vLayoutStyle.setOnClickListener(this)
    }

    override fun onClick(view: View?) {
        when(view?.id) {
            R.id.complex_recyclerview -> openActivity(ComplexRecyclerViewActivity::class.java)
            R.id.v_layout_custom -> openActivity(VCustomActivity::class.java)
            R.id.v_layout_style -> openActivity(VStyleActivity::class.java)
        }
    }

}
